src/core/umount.c

```diff
@@ -25,6 +25,9 @@
 #include <sys/mount.h>
 #include <sys/swap.h>
 
+/* This needs to be after sys/mount.h :( */
+#include <libmount.h>
```

Comment:
@karelzak hey, karel, this seems buggy in libmount.h!

Reply:
Oooops, fixed -- the change will be in util-liux >= 2.32-rc2. Thanks for your report!

Reply:
BTW, just for the record. The old versions of kernel-headers define MS_* macros on two places: linux/fs.h and sys/mount.h. You cannot use the both files together ;-( 

It's already fixed by kernel commit e462ec50cb5fad19f6003a3d8087f4a0945dd2b1 (but fedora 27 still contains old fs.h). Now fs.h uses prefix SB_* and mount.h uses MS_*.  

The libmount.h contains ifndef fallbacks for MS_* macros, and the current (fixed) git tree version includes sys/mount.h now to make sure we do not conflict when included in 3rd party code before sys/mounts.h.

Reply:
Ah, the note about fs.h is not valid. It uses SB_  prefix for internal kernel interfaces, but in userspace (kernel code include/uapi/linux/fs.h) it still uses MS_ prefix. So, the problem is not fixed and fs.h sucks everywhere... has to ask kernel devel about it.

Reply:
This patch https://github.com/karelzak/systemd/commit/eb50b6d936c474f5bc2974c059f6432222af4de4 clean ups the includes to avoid collision between sys/mount.h (and libmount.h) and linux/fs.h. Works. for me. See also https://github.com/systemd/systemd/issues/8507.